composer/loggers/progress_bar_logger.py

```diff
@@ -123,47 +141,65 @@ def should_log(state: State, log_level: LogLevel, console_log_level: LogLevel =
             self.should_log = should_log
         else:
             self.should_log = console_log_level
+
+        # set the stream
         if isinstance(stream, str):
-            if stream.lower() == 'stdout':
-                stream = sys.stdout
-            elif stream.lower() == 'stderr':
-                stream = sys.stderr
-            else:
-                raise ValueError(
-                    f"Invalid stream option: Should be 'stdout', 'stderr', or a TextIO-like object; got {stream}")
+            stream = {
+                'stdout': sys.stdout,
+                'stderr': sys.stderr,
+            }[stream.lower()]
         self.stream = stream
 
-    def log_data(self, state: State, log_level: LogLevel, data: Dict[str, Any]) -> None:
-        if dist.get_local_rank() == 0 and self.is_train in self.pbars:
-            # Logging outside an epoch
-            assert self.is_train is not None
-            self.pbars[self.is_train].log_data(data)
+    @property
+    def current_pbar(self) -> Optional[_ProgressBar]:
+        if self.is_train is None:
+            return None
+        return self.train_pbar if self.is_train else self.eval_pbar
 
-        if not self.should_log(state, log_level):
-            return
-        data_str = format_log_data_value(data)
-        log_str = f'[{log_level.name}][batch={int(state.timestamp.batch)}]: {data_str}'
+    @current_pbar.setter
+    def current_pbar(self, pbar: Optional[_ProgressBar]):
+        if self.is_train is None:
+            raise AssertionError('Cannot set pbar if self.is_train is not set.')
+
+        if self.is_train:
+            self.train_pbar = pbar
+        else:
+            self.eval_pbar = pbar
 
-        if self.is_train in self.pbars:
-            # use tqdm.write to avoid interleaving with a progress bar
-            assert self.is_train is not None
-            self.pbars[self.is_train].pbar.write(log_str)
+    def log_data(self, state: State, log_level: LogLevel, data: Dict[str, Any]) -> None:
+        # log to progress bar
+        if dist.get_local_rank() == 0 and self.current_pbar:
+            # Logging outside an epoch
+            self.current_pbar.log_data(data)
+
+        # log to console
+        if self.should_log(state, log_level):
+            data_str = format_log_data_value(data)
+            log_str = f'[{log_level.name}][batch={int(state.timestamp.batch)}]: {data_str}'
+            self.log_to_console(log_str)
+
+    def log_to_console(self, log_str: str):
+        """Logs to the console, avoiding interleaving with a progress bar."""
+        if self.current_pbar:
+            # use tqdm.write to avoid interleaving
+            self.current_pbar.pbar.write(log_str)
         else:
             # write directly to self.stream; no active progress bar
             print(log_str, file=self.stream, flush=True)
 
+    @rank_zero_only
```

Comment:
On line 171, it is local rank zero -- here, it is global rank zero. Would it make sense for progress bars to show up on all local rank zeros? (Also, `rank_zero` is a bit of a confusing name because of this -- if you do keep the decorator approach, could it be called `_local_rank_zero_only`?)